chart/router/templates/deployment.yaml

```diff
@@ -0,0 +1,85 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "router.fullname" . }}
+  labels:
+    {{- include "router.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      {{- include "router.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "router.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - --hot-reload
+            - --config
+            - /app/configuration.yaml
+          env:
+            {{- if not (empty .Values.managedFederation.apiKey) }}
+            - name: APOLLO_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ include "router.fullname" . }}
+                  key: managedFederationApiKey
+            {{- end }}
+            {{- if not (empty .Values.managedFederation.graphRef) }}
+            - name: APOLLO_GRAPH_REF
+              value: {{ .Values.managedFederation.graphRef }}
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /.well-known/apollo/server-health
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /.well-known/apollo/server-health
+              port: http
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          volumeMounts:
+            - name: router-configuration
+              mountPath: /app/configuration.yaml
+              subPath: configuration.yaml
+              readOnly: true
```

Comment:
This needs to be configurable.

Reply:
Do you mean configurable for the mount path? Or other configurable options?

Reply:
I don't think we should assume that people always want to mount a configuration. I haven't honestly thought through the details, but it feels like we only want to do this if we are hot reloading or specifying a custom configuration so maybe we could tie this all together somehow? I'm very open to suggestions here...

Reply:
I'll leave this as is for now and re-visit in follow up work.